packages/node/src/utils/kyve/kyve.ts

```diff
@@ -0,0 +1,555 @@
+    if (isMainThread) {
+      const files = await fs.promises.readdir(fileCacheDir);
+
+      const isStale = async (file: string) =>
+        ((await fs.promises.stat(file)).mode & 0o777).toString(8) === '200';
+
+      files
+        .filter((f) => KyveApi.isBundleFile(f, poolId))
+        .map(async (bundleFile) => {
+          if (await isStale(bundleFile)) {
+            try {
+              logger.debug(`Removing stale bundle: ${bundleFile}`);
+              await fs.promises.unlink(bundleFile);
+            } catch (e) {
+              if (e.code !== 'ENOENT') {
+                logger.error(`Failed to clear bundle: ${bundleFile}`);
+                throw e;
+              }
+            }
+          }
+        });
+    }
+  }
+
+  private static isBundleFile(filename: string, poolId: string): boolean {
+    return BUNDLE_FILE_ID_REG(poolId).test(filename);
+  }
+
+  private decodeBlock(block: JsonRpcSuccessResponse): BlockResponse {
+    return Responses.decodeBlock({
+      id: 1,
+      jsonrpc: '2.0',
+      result: block,
+    });
+  }
+
+  private decodeBlockResult(
+    blockResult: JsonRpcSuccessResponse,
+  ): BlockResultsResponse {
+    return Responses.decodeBlockResults({
+      id: 1,
+      jsonrpc: '2.0',
+      result: blockResult,
+    });
+  }
+
+  private getBundleFilePath(id: string): string {
+    return path.join(this.tmpCacheDir, `bundle_${this.poolId}_${id}.json`);
+  }
+
+  private async getBundleById(bundleId: number): Promise<BundleDetails> {
+    return (this.cachedBundleDetails[bundleId] ??= (() => {
+      logger.debug(`getBundleId ${bundleId}`);
+      return this.lcdClient.kyve.query.v1beta1.finalizedBundle({
+        pool_id: this.poolId,
+        id: bundleId.toString(),
+      }) as Promise<BundleDetails>;
+    })());
+  }
+
+  private async getResolvedBundleDetails(): Promise<BundleDetails[]> {
+    return Promise.all(Object.values(this.cachedBundleDetails));
+  }
+
+  private async getLatestBundleId(): Promise<number> {
+    return parseDecimal(
+      (
+        await this.lcdClient.kyve.query.v1beta1.finalizedBundles({
+          pool_id: this.poolId,
+          index: '1',
+          pagination: {
+            reverse: true,
+            limit: '1',
+          },
+        })
+      ).finalized_bundles[0].id,
+    );
+  }
+
+  private async bundleIdIterator(height: number): Promise<number> {
+    let bundle: BundleDetails;
+    let bundleId: number;
+
+    const cachedBundles = await this.getResolvedBundleDetails();
+    const nearestBundle = cachedBundles.filter(
+      (b) => parseDecimal(b.to_key) < height,
+    );
+
+    bundleId = Math.max(...nearestBundle.map((b) => parseDecimal(b.id))) + 1;
+
+    bundle = await this.getBundleById(bundleId);
+
+    while (parseDecimal(bundle.to_key) < height) {
+      bundleId++;
+      bundle = await this.getBundleById(bundleId);
+    }
+    return parseDecimal(bundle.id);
+  }
+
+  private async getBundleId(height: number): Promise<number> {
```

Comment:
These names can be more similar, also having a JSDOC description how they differ would be handy